### DISCUSSION TITLE
Mute Not Working on Airplay Player Provider using WiiM Pro Device

### DISCUSSION THREAD

**Discussion by Happy-Cadaver** [music-assistant/support#3371] [Q&A] [answered]
I don't like to open issues until I'm sure it's and issue and not just my lack of understanding, so asking here first.

HA 2024.12.5 and addon MA 2.3.4 with the native integration, same results with HA 2025.1.0bx. Player provider is Airplay using a WiiM Pro device, firmware up to date.

Restarted MA and HA, using developer tools to perform an action.

The mute action does not show the MA created media_player device or entity as an option in the UI, but it does show the automatically created Chromecast media_player entity. Performing the mute action on the Chromecast entity pauses MA media_player obviously via  Chromecast.

When I go to YAML mode and manually enter the HA created media_player entity_id as a target it throws this error in the UI: `Failed to perform the action media_player.volume_mute. Entity media_player.livingroom_stereo does not support this service`.

MA integration debug logging turned on, log output did not change.

HA system logs: 
```
2024-12-26 11:38:34.369 ERROR (MainThread) [homeassistant.helpers.script.websocket_api_script] websocket_api script: Error executing script. Error for call_service at pos 1: Entity media_player.livingroom_stereo does not support this service.
2024-12-26 11:38:34.369 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140302965237664] Error handling message: Entity media_player.livingroom_stereo does not support this service. (home_assistant_error) Rick from 192.168.0.233 (Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.5845.140 Safari/537.36)
```

If I mute from the MA server interface I get this in the server logs, so it’s not just HA related.

`2024-12-26 11:02:20.256 INFO (MainThread) [music_assistant.players] Player Livingroom Stereo does not support muting, using volume instead.`

Thanks in advance for looking at this.

**Comment by OzGav** [comment]
It would appear that the player does not support the mute command. MA is working around it by setting the volume to zero instead.

Seems supported by your statement that HA is working around the problem by pausing the player?

**Reply by Happy-Cadaver** [reply]
HA is not working around it, it simply throws an error and logs it but nothing tangible happens. My thought is also that the WiiM player device is the issue since, as per my question last week in Discord, it seems that Airplay itself supports mute.

My workaround has been to create a helper that records the current volume before seting the volume to 0, and then using that helper to restore the previous volume. I'm using your ha-floorplan based media card and package (updated to work with the new get_queue object) to accomplish this for my personal needs.

However I would have expected that the MA integration would have the same work around as the MA server does, but it does not. Is that a bug or something that can't be done via the native integration?

**Reply by OzGav** [reply]
Yes it would be what the device reports as its capabilities. I don’t have one but it is surprising that mute isn’t supported. You could, out of interest, try the chromecast provider in MA with the device and see if there is any difference there. 

As an aside with the new get_queue action, you may have already noticed this but, the favourite, explicit and disc art functionality is not yet available.

**Reply by OzGav** [reply]
Also just on your last point, when you try and use the pause media player action that is a HA task so it just responds with what the reported player functionality is. A new music_assistant.pause action would have to be created I would assume to mimic the MA functionality…?

**Reply by Happy-Cadaver** [reply, marked as answer]
> You could, out of interest, try the chromecast provider in MA with the device and see if there is any difference there.

Using Chromecast on the same device throughh MA actually mutes it, which I guess suggests that it's the devices implementation of Airplay that is faulty.

> As an aside with the new get_queue action, you may have already noticed this but, the favourite, explicit and disc art functionality is not yet available.

I did, glad to see you add the word 'yet' to that statement :)... also bit rate. That said disc art does work it's just remapped to `image:` under `media_item:` and `album:`

> Also just on your last point, when you try and use the pause media player action that is a HA task so it just responds with what the reported player functionality is. A new music_assistant.pause action would have to be created I would assume to mimic the MA functionality…?

I'm assuming you meant mute and not pause? Pause is working as expected. Otherwise yes I assume MA would have to add that as an action since I can't see HA devs bothering with it.
